### PR TITLE
Fix gallery rendering on page load so current breakpoint configuration isn't replaced by any other configuration

### LINKED ISSUE #29933: Gallery multiply breakpoints options configuration doesn't work properly because of mediaCheck triggering exit media query function on page load
### Preconditions (*)
1. Magento 2.3 / 2.4-develop
2. Multiply breakpoints configured with custom options in `[your theme]/etc/view.xml`

### Steps to reproduce (*)
1. Define more than one breakpoints with custom options

```xml
    <vars module="Magento_Catalog">
        <var name="breakpoints">
            <var name="desktop">
                <var name="conditions">
                    <var name="min-width">1024px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="navdir">vertical</var>
                    </var>
                </var>
            </var>
            <var name="mobile">
                <var name="conditions">
                    <var name="max-width">767px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="nav">dots</var>
                    </var>
                </var>
            </var>
        </var>
    </vars>
```

2. Set mobile screen size
3. Load page

### Expected result (*)
Gallery should have "dots" navigation.

### Actual result (*)
Gallery have "thumbs" navigation.

**Severity**: S2 - S4

### PR DESCRIPTION
**Pull request by Januszpl**
### Description (*)
Fixes:  #29933 

If we have custom configuration for more than one defined breakpoint in `etc/view.xml` current gallery js module triggers matchMedia method for every breakpoint. This method triggers entry or exit function on page load for every defined breakpoint so it can replace previous configuration. For example we load page on mobile and it first runs entry function for current breakpoint but will also run exit function for desktop breakpoint if defined. This way we can have wrong looking gallery.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#29933

### Manual testing scenarios (*)
1. Set in your theme (etc/view.xml)

On default Luma settings we won't see this issue because configuration has only one breakpoint defined.
`"breakpoints": {"mobile":{"conditions":{"max-width":"767px"},"options":{"options":{"nav":"dots"}}}}            }`

So we need have this:

```xml
var name="breakpoints">
            <var name="desktop">
                <var name="conditions">
                    <var name="min-width">1024px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="navdir">vertical</var>
                    </var>
                </var>
            </var>
            <var name="mobile">
                <var name="conditions">
                    <var name="max-width">767px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="nav">dots</var>
                    </var>
                </var>
            </var>
        </var>
    </vars>
```

2. Load product page

3. On page load on mobile dot nav should be visible

### Questions or comments
This is quick fix (limited time). Perhaps it could be fixed in other way. Please verify. If you will notice better option let me know I will try to check it again.

### CODE REVIEW

**Comment by Januszpl** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @Januszpl. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @Januszpl, here is your [new Magento instance](https://pr-29934.instances.magento-community.engineering).
Admin access: [https://pr-29934.instances.magento-community.engineering/admin_c2a4](https://pr-29934.instances.magento-community.engineering/admin_c2a4)
Login: `fff453a3` Password: `81958b660945`
Instance will be terminated in up to 3 hours.

**Comment by Januszpl** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @Januszpl. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Januszpl, here is your [Magento instance](https://i-29934-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-29934-2-4-develop.instances.magento-community.engineering/admin_22c9](https://i-29934-2-4-develop.instances.magento-community.engineering/admin_22c9)
Login: `0d8e0d60` Password: `0ed4a9e3e4f6`
Instance will be terminated in up to 3 hours.

File: lib/web/mage/gallery/gallery.js [hunk 1]
```diff
@@ -339,29 +342,37 @@ define([
...
                             }
 
-                            settings.api.updateOptions(settings.breakpoints[pair[0]].options, true);
-                            $.extend(true, config, settings.breakpoints[pair[0]]);
-                            settings.activeBreakpoint = settings.breakpoints[pair[0]];
+                            settings.api.updateOptions(settings.breakpoints[mediaQuery].options, true);
+                            $.extend(true, config, settings.breakpoints[mediaQuery]);
+                            settings.activeBreakpoint = settings.breakpoints[mediaQuery];
+
+                            if (!isInitialized[mediaQuery]) {
```
Comment (rogyar) [inline lib/web/mage/gallery/gallery.js]: I would recommend removing this check (condition). Simply `isInitialized[mediaQuery] = true;` will be more sufficient from the performance and code readability perspective 
```suggestion

```

**Comment by rogyar** [comment]
Hi @Januszpl. Thank you for your collaboration. Great job! 
Please, check my minor recommendation above.

**Comment by rogyar** [comment]
Also, according to the [definition of done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html), all changes should be covered by automated tests. It would be awesome if you could extend the [existing test](https://github.com/rogyar/magento2/blob/acdc9b429bfab2a9aa496f571d9ccc37d0b6492d/dev/tests/js/jasmine/tests/lib/mage/gallery/gallery.test.js#L7) to cover your changes. 

Thank you!

File: lib/web/mage/gallery/gallery.js (hunk 1 above)
Comment (Januszpl) [reply lib/web/mage/gallery/gallery.js]: Thanks for review. Change pushed.

**Comment by Januszpl** [comment]
@rogyar I will try to check if I can extend gallery and I will let you know by the end of this week.

**Comment by slavvka** [comment]
@magento give me test instance
with edition B2B

**Comment by slavvka** [comment]
@magento give me new test instance
with edition EE

**Comment by gabrieldagama** [comment]
@magento run all tests

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @Januszpl, thanks for the contribution, it looks good! Please let us know if you need assistance on the test coverage, we can help you with it.

Thanks.

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Review by gabrieldagama — APPROVED** [review]
Thanks for adding test coverage, the changes are approved.

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8183](https://jira.corp.magento.com/browse/ENGCOM-8183) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by Januszpl** [comment]
I just put all changes in one commit.

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8183](https://jira.corp.magento.com/browse/ENGCOM-8183) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
**Result:**
:x: **Before:**
Media gallery thumbs do not switch to dots on mobile view
29934Main


:heavy_check_mark: **After:**
Media gallery thumbs is displayed as dots on mobile view
#29934PR

**Comment by engcom-Delta** [comment]
:warning: **Note:** Automation tests in waiting status

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Pull request merged**